lib/archethic/p2p/client/connection.ex

```diff
@@ -77,8 +77,20 @@ defmodule Archethic.P2P.Client.Connection do
       availability_timer: {nil, 0}
     }
 
-    actions = [{:next_event, :internal, :connect}]
-    {:ok, :disconnected, data, actions}
+    # First connection attempt is synchronous (to deal with the `start_link` followed by a `send_message`)
+    # We call ourselves the `state_enter` because `GenStateMachine` will not do it from init
+    # `handle_connect` might take a long time
+    case transport.handle_connect(ip, port) do
+      {:ok, socket} ->
+        {_, new_data} = handle_event(:enter, :disconnected, {:connected, socket}, data)
+        {:ok, {:connected, socket}, new_data}
+
+      {:error, _} ->
+        {_, new_data, actions} =
+          handle_event(:enter, {:connected, make_ref()}, :disconnected, data)
```

Comment:
I think here we should not call this handle event as we where not previously in `:connected` state. We should just return
```elixir
actions = [{{:timeout, :reconnect}, 500, nil}]
{:ok, :disconnected, data, actions}
```